File: 1ºAno/Lista3/C/Sala/files/ex11.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main() {
    for (;;) {
        char *nums = (char*)malloc(100 * sizeof (char));
        printf ("Na mesma linha, entre com os números inteiros, que eu irei dizer qual é o maior e o menor número\nPara sair, entre com 0\n\n");
        fgets (nums, 100, stdin);
        char *pt = strtok(nums, " ");
        int maior = atoi(pt), menor = atoi(pt);
        while (pt != NULL) {
            if (atoi(pt) > maior) {
                maior = atoi(pt);
            }
            if (atoi(pt) < menor) {
                menor = atoi(pt);
            }
            pt = strtok(NULL, " ");
        }
        if (maior == 0 && menor == 0) {
            break;
        }
        printf ("Maior valor = %d\nMenor valor = %d\n\n", maior, menor);
    }
    printf ("\nAdeus");
    return 0;
}

File: 1ºAno/Lista3/Python/Sala/files/ex11.py
while True:
    while True:
        try:
            nums = input("Na mesma linha, entre com os números que desejar\nIrei dizer qual é o maior e o menor\nPara sair, entre com apenas um 0\n\n").split()
            nums = [int(i) for i in nums]
            break
        except ValueError:
            print ("Número(s) inválido(s)... Tente novamente")
    if len(nums) == 1 and nums[0] == 0:
        break
    print ("Maior = {}\nMenor = {}\n" .format(max(nums), min(nums)))
print ("\nAdeus!")